### PR TITLE
[WIP] Introduced `Validator::format()`

### PR DESCRIPTION
**Pull request by SamMousa**
Refactored all validators to use `$this->format` instead of `Yii::$app->getI18n()->format`.

| Q             | A
| ------------- | ---
| Is bugfix?    | yes (subjective)
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | 
| Fixed issues  | #13254

### CODE REVIEW

File: framework/validators/Validator.php [hunk 1]
```diff
@@ -432,4 +432,25 @@ public function isEmpty($value)
             return $value === null || $value === [] || $value === '';
         }
     }
+
+    /**
+     * Formats a mesage using the I18N, or simple strtr if \Yii::$app is not available.
+     * Same implementation as used by \Yii::t
+     * @param string $message
+     * @param array $params
+     * @return string
```
Comment (samdark) [inline framework/validators/Validator.php]: Need `@since` tag.

"Same implementation as used by \Yii::t" isn't needed here.

**Review by samdark — CHANGES_REQUESTED** [review]

**Review by dynasource — APPROVED** [review]
nice cleanup. I'd tend to prefer a method name `formatMessage` as its more descriptive

File: framework/validators/Validator.php [hunk 2]
```diff
@@ -432,4 +432,25 @@ public function isEmpty($value)
             return $value === null || $value === [] || $value === '';
         }
     }
+
+    /**
+     * Formats a mesage using the I18N, or simple strtr if \Yii::$app is not available.
```
Comment (SilverFire) [inline framework/validators/Validator.php]: `\Yii::$app` -> <code>\`\Yii::$app\`</code>

File: framework/validators/Validator.php [hunk 3]
```diff
@@ -432,4 +432,25 @@ public function isEmpty($value)
...
+     * Same implementation as used by \Yii::t
```
Comment (SilverFire) [inline framework/validators/Validator.php]: `\Yii::t` -> <code>\`\Yii::t\`</code>

**Review by SilverFire — CHANGES_REQUESTED** [review]
Good PR

**Comment by SamMousa** [comment]
Will fix this tomorrow.

**Comment by SamMousa** [comment]
@dynasource I chose `format` since it is used by formatters as well.
Let me know if I must change it to `formatMessage`.

**Comment by samdark** [comment]
`formatMessage` looks better to me in this case since context isn't clear from just `format`.

**Review by samdark — APPROVED** [review]

**Comment by SamMousa** [comment]
That refactor will have to wait till tomorrow so i can do it the lazy way with an IDE :)

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**